backend/src/core/annotation.py

```diff
@@ -66,62 +68,59 @@ def process_tasks(annotation_queue, genomic_unit_collection):  # pylint: disable
 
                 if not ready:
-                    delay_count = dataset_json['delay_count'] + 1 if 'delay_count' in dataset_json else 0
-                    dataset_json['delay_count'] = delay_count
+                    annotation_unit.set_delay_count()
 
-                    if dataset_json['delay_count'] < 10:
+                    if annotation_unit.check_delay_count():
                         logger.info(
                             '%s Delaying Annotation, Missing Dependency...',
-                            annotation_unit_string(genomic_unit['unit'], dataset_json['data_set'])
+                            annotation_unit_string(annotation_unit.get_genomic_unit(), annotation_unit.get_dataset())
                         )
-                        annotation_queue.put((genomic_unit, dataset_json))
+                        annotation_queue.put(annotation_unit)
                     else:
-                        missing_dependencies = [
-                            dependency for dependency in dataset_json['dependencies'] if dependency not in genomic_unit
-                        ]
+                        missing_dependencies = annotation_unit.get_missing_dependencies()
                         logger.info(
                             '%s Canceling Annotation, Missing %s ...',
-                            annotation_unit_string(genomic_unit['unit'], dataset_json['data_set']), missing_dependencies
+                            annotation_unit_string(annotation_unit.get_genomic_unit(), annotation_unit.get_dataset()),
+                            missing_dependencies
                         )
 
                     continue
 
-                task = AnnotationTaskFactory.create(genomic_unit, dataset_json)
+                task = AnnotationTaskFactory.create(annotation_unit.genomic_unit, annotation_unit.dataset)
                 logger.info(
                     '%s Creating Task To Annotate...',
-                    annotation_unit_string(genomic_unit['unit'], dataset_json['data_set'])
+                    annotation_unit_string(annotation_unit.get_genomic_unit(), annotation_unit.get_dataset())
                 )
 
-                annotation_task_futures[executor.submit(task.annotate)] = (genomic_unit, task)
+                annotation_task_futures[executor.submit(task.annotate)] = (annotation_unit.genomic_unit, task)
 
                 for future in concurrent.futures.as_completed(annotation_task_futures):
-                    genomic_unit, annotation_task = annotation_task_futures[future]
+                    annotation_unit.genomic_unit, annotation_task = annotation_task_futures[future]
                     logger.info(
-                        '%s Query completed...', annotation_unit_string(genomic_unit['unit'], dataset_json['data_set'])
+                        '%s Query completed...',
+                        annotation_unit_string(annotation_unit.get_genomic_unit(), annotation_unit.get_dataset())
```

Comment:
Ok, so after taking a look at this, I think this can go a step further and we can remove the neccessity of the `annotation_logging.py` file. https://github.com/uab-cgds-worthey/rosalution/blob/main/backend/src/core/annotation_logging.py


`{genomic_unit} for {data_set}` can get moved into the 'AnnotationUnit' class as a function that is like 'to_name_string()'.,  and the other parts of the annotation_logging.py file can be moved to be a function inside 'annotation.py' to be 'format_annotation_logging(annotation_unit)'.  Let me know if this makes sense and I can talk through it more with you.

Reply:
I made these changes. Please let me know if this looks good, and then I can delete the `annotation_logging` file

Reply:
Looks great! I will go aheand and resolve this.